pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go

```diff
@@ -316,3 +321,72 @@ func TestCutoverBuiltin(t *testing.T) {
 	require.True(t, ok)
 	require.Equal(t, hlc.Timestamp{WallTime: highWater.UnixNano()}, sp.StreamIngest.CutoverTime)
 }
+
+// BenchmarkCutover benchmarks issueRevertRangeRequests, the function which the
+// stream ingestion job will use to revert ranges to a specified cutover time.
+// For a given benchmark trial, the following occurs:
+//
+// 1. Insert X number of rows with random keys into the cluster, using the bank workload
+// 2. Issue Y splits
+// 3. Record a cutover time
+// 4. Insert another Z number of rows with random keys into the cluster
+// 5. Issue revert ranges on the bank table to the cutover time.
+func BenchmarkCutover(b *testing.B) {
+	defer leaktest.AfterTest(b)()
+	defer log.Scope(b).Close(b)
+	ctx := context.Background()
+
+	const payloadSize = 100
+
+	rowNums := []int{1000, 10000}
+	for _, initRows := range rowNums {
+		for _, splits := range []int{0, 10} {
+			if splits >= initRows {
+				continue
+			}
+			for _, cutoverRows := range rowNums {
+				if cutoverRows > initRows {
+					continue
+				}
+				subtestName := fmt.Sprintf("initRows=%d;numSplits=%d;cutoverRows=%d", initRows, splits, cutoverRows)
+				b.Run(subtestName, func(b *testing.B) {
+
+					tc := testcluster.StartTestCluster(b, 3, base.TestClusterArgs{})
```

Comment:
i truly wish we didn't need to spin up a new test cluster for every iteration, but i'm not sure there's a clean way to reset the cluster. I obviously could clearRange everything and merge all the ranges, but I imagine there's tons of caches and background processes still going on. I wish the testcluster had a Reset() method.